dvd/dvd_dump.ixx

```diff
@@ -412,11 +430,131 @@ export bool redumper_dump_dvd(Context &ctx, const Options &options, DumpMode dum
                 physical_sectors_count += get_layer_length(layer_descriptor);
             }
 
-            // Kreon drives return incorrect sectors count
             if(physical_sectors_count != sectors_count)
+            {
+                // Kreon PFI sector count is only for Video portion when XGD present
+                if(ctx.drive_config.vendor_specific.starts_with("KREON V1.00"))
+                    is_xbox = true;
+                else
+                {
+                    LOG("warning: READ_CAPACITY / PHYSICAL sectors count mismatch, using PHYSICAL");
+                    sectors_count = physical_sectors_count;
+                }
+            }
+        }
+
+        uint32_t xbox_layer0_end_sector = 0;
+        if(is_xbox)
+        {
+            std::vector<uint8_t> security_sector(0x800);
+
+            status = cmd_kreon_get_security_sector(*ctx.sptd, security_sector);
+            if(status.status_code)
+                throw_line("failed to get security sectors, SCSI ({})", SPTD::StatusMessage(status));
+
+            // store security sector
+            if(dump_mode == DumpMode::DUMP)
+                write_vector(image_prefix + ".raw_ss", security_sector);
+
+            // validate security sector
+            XGD_Type xgd_type = get_xgd_type(security_sector);
+            if(xgd_type == XGD_Type::UNKNOWN)
             {
                 LOG("warning: READ_CAPACITY / PHYSICAL sectors count mismatch, using PHYSICAL");
-                sectors_count = physical_sectors_count;
+                LOG("warning: Kreon Drive with malformed XGD detected, reverting to normal DVD mode");
+                LOG("");
+                is_xbox = false;
+            }
+
+            if(is_xbox && !physical_structures.empty())
+            {
+                LOG("Kreon Drive with XGD{} detected", (uint8_t)xgd_type);
+                LOG("");
+
+                std::vector<uint8_t> clean_ss = security_sector;
+                clean_xbox_security_sector(clean_ss);
```

Comment:
The `.raw_ss` is the security sector as received by the drive, this cleaned version is the one that is used for redump submissions as (I might get some details wrong here) my understanding is the security sector contains disc angle data, and if there was a new manufacturing run of a disc with the same data the security sector would have different angles in it. This cleaned version overwrites that angle data in order to make sure that we can check if the rest of the security sector is the same. @Deterous do you happen to know more details about this?

For the non-dump runs though we can check the `.raw_ss` if you think that would be better. The goal of doing the cleaned version for the comparison is if you have two of the same disc you would be able to verify that the contents of the disc are the same even if it has different angles in the security sector.

Reply:
I may has misinterpreted this, if you are referring to the variable itself, yes I think we can reuse the `security_sector` variable and don't need to copy to a new vector

Reply:
Reason for clean_ss is both for making sure we refine the same disc as you mentioned, and also for redump. The clean SS is the hash that redump puts in comments, and it will match the output of ss_sector_range (dumpers won't have to run that program anymore).
We had to put it in `redumper dump` code because we use it for checking same disc is being refined as you mentioned.

Reply:
Ok, so we have two things:
1. raw security sector dump that will have different angle data for the discs with otherwise same dump data
2. security sector dump that has angle data zeroed and redump is using hash of this file in the submission

Specifically, I don't like outputting both files with otherwise identical content when using `redumper dump` command. This command should always store only what was received from the drive in the most RAW formats. 

Here's what I would like to do here:
(1) should be dumped as .security
(2) should be generated from (1) on refine in memory and compared to the one from disc to ensure we refine the same disc

Hash for the cleaned one should be printed in `redumper info` command from (1) when it detects xbox dump.

Optionally, (2) can be generated during `redumper info` using a command line option. We have a similar problem with PSX discs where we want to generate .sbi file for libcrypt games (generated from raw .subcode). I have to think what would be the proper implementation of this.

Reply:
So just to make sure I understand everything correctly:

> (1) should be dumped as .security
* Always save (1) for every dump
> (2) should be generated from (1) on refine in memory and compared to the one from disc to ensure we refine the same disc
* When doing a refine create (2) for the existing `.security` file and for the current disc to be able to do the comparison (but don't save anything to disc, only exists in memory for the given run)
> Hash for the cleaned one should be printed in redumper info command from (1) when it detects xbox dump.
* Create (2) during info to print hash (but don't save as file to disc)

Reply:
> When doing a refine create (2) for the existing .security file and for the current disc to be able to do the comparison (but don't save anything to disc, only exists in memory for the given run)

Yes, basically load .security file to memory, clear angle data, read security sector from the disc, clear it's angle data and compare the two.

> Create (2) during info to print hash (but don't save as file to disc)

That should be the output of `redumper info` command that analyzes dump files and do not access the drive at all. If xbox dump is detected, read .security file, clear angle data in memory, print hash and that's it. 

That should come as a separate PR after we're done with this one.

Reply:
Okay great, that is pretty straight forward. Looking at the code I have one more question.

Currently only one user has reported this happening (so I think it should be uncommon) but their drive sometimes fails to send the full SS sequence leading to only a partial SS being received, in the current code I save it as the `.security` and just log that it is incomplete but didn't perform the cleanup. Now that we are only outputting the `.security` file should I:

a) Continue to save it anyways
b) Save it but change the extension to something like `.partial_security`
c) Do NOT save it (should we attempt to continue the dump (theoretically should be fine) or force a failure)

note in all of these cases it would log that it was incomplete

Reply:
Apologies for the lack of response the past few months, I got caught up with some stuff and then never revisited this.

I believe I have updated the behavior to address everything so now it will:

- Save "raw" SS as `.security` file
- Clean SS to continue using but do not save to file
- When refining (and not forcing) read saved `.security` and clean it so we can do a proper compare without worrying about faked angles

I re-tested with an XGD1 and XGD3 disc and both dumped correctly. I deleted some random portion of `.state` for the XGD1 after it completed and did a refine and it completed and the SHA matched (did a manual SHA-1 compute of the ISO after refine)